src/audio/data_blob.c

```diff
@@ -476,10 +476,8 @@ comp_data_blob_handler_new_ext(struct comp_dev *dev, bool single_blob,
 
 void comp_data_blob_handler_free(struct comp_data_blob_handler *blob_handler)
 {
-	if (!blob_handler)
-		return;
-
-	comp_free_data_blob(blob_handler);
-
-	rfree(blob_handler);
+	if (blob_handler) {
+		comp_free_data_blob(blob_handler);
+		rfree(blob_handler);
+	}
```

Comment:
sorry, I don't think this is a good patch or at least a good commit description. I understand that compilers have their problems and sometimes small and seemingly unrelated changes might change the compiler behaviour, but I don't think we should rely on that. I don't think we can rely on apparent empiric improvements which have no logical explanation. If our success rate depends on the if-else ordering then (arguably) we are doing something wrong. I'd propose to drop this commit and investigate the problem.

Reply:
yes, sorry this patch needs to go. It was there for a test for xtos and not needed with zephyr.